File: 3412-permutation-difference-between-two-strings/permutation-difference-between-two-strings.dart
class Solution {
  int findPermutationDifference(String s, String t) {
    int n = s.length;
    int difference = 0;
    
    Map<String, int> sIndexMap = {};
    for (int i = 0; i < n; i++) {
      sIndexMap[s[i]] = i;
    }
    
    for (int i = 0; i < n; i++) {
      difference += (sIndexMap[t[i]]! - i).abs();
    }
    
    return difference;
  }
}

File: 3412-permutation-difference-between-two-strings/permutation-difference-between-two-strings.rs
impl Solution {
    pub fn find_permutation_difference(s: String, t: String) -> i32 {
        let mut difference = 0;
        let mut s_index_map = [0; 26];

        for (i, c) in s.chars().enumerate() {
            s_index_map[(c as u8 - 'a' as u8) as usize] = i;
        }

        for (i, c) in t.chars().enumerate() {
            difference += (s_index_map[(c as u8 - 'a' as u8) as usize] as i32 - i as i32).abs();
        }

        difference
    }
}